changelog/1587.doc.rst

```diff
@@ -0,0 +1,4 @@
+Increased the strictness of the ``build_docs`` tox_ environment so that
+broken reST_ links now emit warnings which are then treated as errors,
+removed the ``build_docs_nitpicky`` tox_ environment, and updated the
+|documentation guide| accordingly.
```

Comment:
Possibly add something about fixing all links that were failing under `nitpicky`.  This could be added to this entry or as its own.